src/Symfony/Component/HttpFoundation/Tests/Session/Storage/Handler/MongoDbSessionHandlerTest.php

```diff
@@ -54,13 +58,39 @@ protected function setUp(): void
             'collection' => 'session-test',
         ];
 
-        $this->storage = new MongoDbSessionHandler($this->mongo, $this->options);
+        $this->storage = new MongoDbSessionHandler($this->manager, $this->options);
+    }
+
+    public function testCreateFromClient()
+    {
+        $client = $this->createMock(Client::class);
+        $client->expects($this->once())
+            ->method('getManager')
+            ->willReturn($this->manager);
+
+        $this->storage = new MongoDbSessionHandler($client, $this->options);
+        $this->storage->write('foo', 'bar');
+
+        $this->assertCount(1, $this->getSessions());
     }
 
-    public function testConstructorShouldThrowExceptionForMissingOptions()
+    protected function tearDown(): void
+    {
+        $this->manager->executeCommand('sf-test', new Command(['drop' => 'session-test']));
```

Comment:
Note that the `drop` command may fail if the collection doesn't exist. Since this is a general `tearDown()` method, that could be problematic if some test was added to this file and didn't actually write to the collection to implicitly create it.

Consider suppressing the relevant exception code as is done in PHPLIB's [DropCollection](https://github.com/mongodb/mongo-php-library/blob/master/src/Operation/DropCollection.php#L108).

Looking at `testConstructorShouldThrowExceptionForMissingOptions()` below, I'm surprised that this doesn't already fail since that test never gets far enough to write to the collection.

Reply:
The no-op behavior was only recently implemented for collections in server 7.0 ([SERVER-43894](https://jira.mongodb.org/browse/SERVER-43894)). For databases, it was implemented in server 5.0 ([SERVER-55145](https://jira.mongodb.org/browse/SERVER-55145)).

Reply:
Added the try...catch for compatibility with mongodb 6 and lower.